src/pages/api/modules/usenet/nzbget/types.ts

```diff
@@ -0,0 +1,135 @@
+  FileSizeMB: number;
+  FileCount: number;
+  RemainingFileCount: number;
+  MinPostTime: number;
+  MaxPostTime: number;
+  TotalArticles: number;
+  SuccessArticles: number;
+  FailedArticles: number;
+  Health: number;
+  DownloadedSizeLo: number;
+  DownloadedSizeHi: number;
+  DownloadedSizeMB: number;
+  DownloadTimeSec: number;
+  PostTotalTimeSec: number;
+  ParTimeSec: number;
+  RepairTimeSec: number
+  UnpackTimeSec: number;
+  MessageCount:number;
+  DupeKey: string;
+  DupeScore: number;
+  DupeMode: 'SCORE' | 'ALL' | 'FORCE';
+  Status: string;
+  ParStatus: 'NONE' | 'FAILURE' | 'REPAIR_POSSIBLE' | 'SUCCESS' | 'MANUAL';
+  ExParStatus: 'RECIPIENT' | 'DONOR';
+  UnpackStatus: 'NONE' | 'FAILURE' | 'SPACE' | 'PASSWORD' | 'SUCCESS';
+  UrlStatus: 'NONE' | 'SUCCESS' | 'FAILURE' | 'SCAN_SKIPPED' | 'SCAN_FAILURE';
+  ScriptStatus: 'NONE' | 'FAILURE' | 'SUCCESS';
+  ScriptStatuses: [];
+  MoveStatus: 'NONE' | 'SUCCESS' | 'FAILURE';
+  DeleteStatus: 'NONE' | 'MANUAL' | 'HEALTH' | 'DUPE' | 'BAD' | 'SCAN' | 'COPY';
+  MarkStatus: 'NONE' | 'GOOD' | 'BAD';
+  ExtraParBlocks: number;
+  Parameters: [];
+  ServerStats: [];
+}
+
+export interface NzbgetQueueItem {
+  NZBID: number;
+  NZBFilename: string;
+  NZBName: string;
+  Kind: 'NZB' | 'URL';
+  URL: string;
+  DestDir: string;
+  FinalDir: string;
+  Category: string;
+  FileSizeLo: number;
+  FileSizeHi: number;
+  FileSizeMB: number;
+  RemainingSizeLo: number;
+  RemainingSizeHi: number;
+  RemainingSizeMB: number;
+  PausedSizeLo: number;
+  PausedSizeHi: number;
+  PausedSizeMB: number;
+  FileCount: number;
+  RemainingFileCount: number;
+  RemainingParCount: number;
+  MinPostTime: number;
+  MaxPostTime: number;
+  MaxPriority: number;
+  ActiveDownloads: number;
+  Status: 'QUEUED' | 'PAUSED' | 'DOWNLOADING' | 'FETCHING' | 'PP_QUEUED' | 'LOADING_PARS' | 'VERIFYING_SOURCES' | 'REPAIRING' | 'VERIFYING_REPAIRED' | 'RENAMING' | 'UNPACKING' | 'MOVING' | 'EXECUTING_SCRIPT' | 'PP_FINISHED';
+  TotalArticles: number;
+  SuccessArticles: number;
+  FailedArticles: number;
+  Health: number;
+  CriticalHealth: number;
+  DownloadedSizeLo: number;
+  DownloadedSizeHi: number;
+  DownloadedSizeMB: number;
+  DownloadTimeSec: number;
+  MessageCount: number;
+  DupeKey: string;
+  DupeScore: number;
+  DupeMode: string;
+  Parameters: [];
+  ServerStats: [];
+  PostInfoText: string;
+  PostStageProgress: number;
+  PostTotalTimeSec: number;
+  PostStageTimeSec: number;
+}
+
+export interface NzbgetStatus {
+  RemainingSizeLo: number;
+  RemainingSizeHi: number;
+  RemainingSizeMB: number;
+  ForcedSizeLo: number;
+  ForcedSizeHi: number;
+  ForcedSizeMB: number;
+  DownloadedSizeLo: number;
+  DownloadedSizeHi: number;
+  DownloadedSizeMB: number;
+  ArticleCacheLo: number;
+  ArticleCacheHi: number;
+  ArticleCacheMB: number;
+  DownloadRate: number;
+  AverageDownloadRate: number;
+  DownloadLimit: number;
+  ThreadCount: number;
+  PostJobCount: number;
+  UrlCount: number;
```

Comment:
Do we really need stuff like this? I appreciate the type definition's, but it is kind of unnecessary, when we only use a few of them

Reply:
This comment makes no sense lol 😄 This it the output from the NZBGet api, it's what's returned. You want to alter the types to not reflect reality just because it seems excessive?

Reply:
No, you are right - We should keep them. I didn't think this quite through, sorry.